nemo/collections/asr/models/sortformer_diar_models.py

```diff
@@ -0,0 +1,578 @@
+        """
+        self._accuracy_test = MultiBinaryAccuracy()
+        self._accuracy_train = MultiBinaryAccuracy()
+        self._accuracy_valid = MultiBinaryAccuracy()
+
+        self._accuracy_test_ats = MultiBinaryAccuracy()
+        self._accuracy_train_ats = MultiBinaryAccuracy()
+        self._accuracy_valid_ats = MultiBinaryAccuracy()
+
+    def _reset_train_metrics(self):
+        self._accuracy_train.reset()
+        self._accuracy_train_ats.reset()
+
+    def _reset_valid_metrics(self):
+        self._accuracy_valid.reset()
+        self._accuracy_valid_ats.reset()
+
+    def __setup_dataloader_from_config(self, config):
+        # Switch to lhotse dataloader if specified in the config
+        if config.get("use_lhotse"):
+            return get_lhotse_dataloader_from_config(
+                config,
+                global_rank=self.global_rank,
+                world_size=self.world_size,
+                dataset=LhotseAudioToSpeechE2ESpkDiarDataset(cfg=config),
+            )
+
+        featurizer = WaveformFeaturizer(
+            sample_rate=config['sample_rate'], int_values=config.get('int_values', False), augmentor=self.augmentor
+        )
+
+        if 'manifest_filepath' in config and config['manifest_filepath'] is None:
+            logging.warning(f"Could not load dataset as `manifest_filepath` was None. Provided config : {config}")
+            return None
+
+        logging.info(f"Loading dataset from {config.manifest_filepath}")
+
+        if self._trainer is not None:
+            global_rank = self._trainer.global_rank
+        else:
+            global_rank = 0
+        time_flag = time.time()
+        logging.info("AAB: Starting Dataloader Instance loading... Step A")
+
+        dataset = AudioToSpeechE2ESpkDiarDataset(
+            manifest_filepath=config.manifest_filepath,
+            soft_label_thres=config.soft_label_thres,
+            session_len_sec=config.session_len_sec,
+            num_spks=config.num_spks,
+            featurizer=featurizer,
+            window_stride=self._cfg.preprocessor.window_stride,
+            global_rank=global_rank,
+            soft_targets=config.soft_targets if 'soft_targets' in config else False,
+        )
+        logging.info(
+            f"AAB: Dataloader dataset is created, starting torch.utils.data.Dataloader"
+            f"step B: {time.time() - time_flag}"
+        )
+
+        self.data_collection = dataset.collection
+        self.collate_ds = dataset
+
+        dataloader_instance = torch.utils.data.DataLoader(
+            dataset=dataset,
+            batch_size=config.batch_size,
+            collate_fn=self.collate_ds.eesd_train_collate_fn,
+            drop_last=config.get('drop_last', False),
+            shuffle=False,
+            num_workers=config.get('num_workers', 1),
+            pin_memory=config.get('pin_memory', False),
+        )
+        logging.info(f"AAC: Dataloader Instance loading is done ETA Step B done: {time.time() - time_flag}")
```

Comment:
is this info necessary? so many infos overall? Can you check if you can move some of them to warnings or debugs?

Reply:
This is not needed. I guess this one is accidentally survived from the debugging code. Removed.